src/RelationViewer/RelationViewerProvider.ts

```diff
@@ -0,0 +1,192 @@
+    ]);
+
+    registrations.forEach(disposable => context.subscriptions.push(disposable));
+  }
+
+  constructor(private readonly context: vscode.ExtensionContext) {
+    this._context = context;
+  }
+
+  // CustomReadonlyEditorProvider implements
+  async openCustomDocument(
+      uri: vscode.Uri, openContext: {backupId?: string},
+      _token: vscode.CancellationToken): Promise<RelationViewerDocument> {
+    const document: RelationViewerDocument = await RelationViewerDocument.create(uri);
+    // NOTE as a readonly viewer, there is not much to do
+
+    // TODO handle dispose
+    // TODO handle file change events
+    // TODO handle backup
+
+    return document;
+  }
+
+  // CustomReadonlyEditorProvider implements
+  async resolveCustomEditor(
+      document: RelationViewerDocument, webviewPanel: vscode.WebviewPanel,
+      _token: vscode.CancellationToken): Promise<void> {
+    document.openView(webviewPanel, this._context.extensionUri, document.uri);
+  }
+}
+
+export function getRelationData(path:any) {
+  return {
+    "selected": "1",
+    "relationData": [
+      {"id": "1", "parent": "", "idx": 0, "dataList": [{"name": "a.tflite", "path": "./a.tflite"}]},  // TODO: id, parentid: hashId
+      {"id": "2", "parent": "1", "idx": 0, "dataList": [{"name": "b.circle", "path": "./b.circle", "oneccVersion": "1.0.0", "toolchainVersion": "1.0.0"}]},
+      {"id": "3", "parent": "1", "idx": 0, "dataList": [{"name": "c.circle", "path": "./c.circle", "oneccVersion": "1.0.0", "toolchainVersion": "1.0.0"}]},
+      {"id": "4", "parent": "2", "idx": 0, "dataList": [{"name": "b1.circle", "path": "./b1.circle", "oneccVersion": "1.0.0", "toolchainVersion": "1.0.0"}]},
+      {"id": "5", "parent": "2", "idx": 0, "dataList": [{"name": "b2.circle", "path": "./b2.circle", "oneccVersion": "1.0.0", "toolchainVersion": "1.0.0"}]},
+      {"id": "6", "parent": "2", "idx": 0, "dataList": [{"name": "b3.circle", "path": "./b3.circle", "oneccVersion": "1.0.0", "toolchainVersion": "1.0.0"}]},
+      {"id": "7", "parent": "2", "idx": 0, "dataList": [{"name": "b4.circle", "path": "./b4.circle", "oneccVersion": "1.0.0", "toolchainVersion": "1.0.0"}]},
+      {"id": "8", "parent": "2", "idx": 0, "dataList": [{"name": "b5.circle", "path": "./b5.circle", "oneccVersion": "1.0.0", "toolchainVersion": "1.0.0"}]},
+      {"id": "9", "parent": "3", "idx": 0, "dataList": [{"name": "d.circle", "path": "./d.circle", "oneccVersion": "1.0.0", "toolchainVersion": "1.0.0"}]},
+      {"id": "10", "parent": "9", "idx": 0, "dataList": [{"name": "e.circle", "path": "./e.circle", "oneccVersion": "1.0.0", "toolchainVersion": "1.0.0"}]},
+      {"id": "11", "parent": "10", "idx": 0, "dataList": [{"name": "e1.circle", "path": "./e1.circle", "oneccVersion": "1.0.0", "toolchainVersion": "1.0.0"}]},
+      {"id": "12", "parent": "10", "idx": 0, "dataList": [{"name": "e2.circle", "path": "./e2.circle", "oneccVersion": "1.0.0", "toolchainVersion": "1.0.0"}]}
+    ]
+  };
+}
```

Comment:
Could you add a line at the end of the file?

Reply:
Oh, I'll apply that to all the files!